src/components/CollectionsPage/CollectionsPage.tsx

```diff
@@ -87,17 +108,20 @@ export default function CollectionsPage(props: Props) {
     }
   }, [onOpenModal, isLinkedWearablesPaymentsEnabled])
 
-  const handleSearchChange = (_evt: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => {
-    setSearch(data.value)
-    if (timeout.current) {
-      clearTimeout(timeout.current)
-      timeout.current = null
-    }
+  const handleSearchChange = useCallback(
+    (_evt: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => {
+      setSearch(data.value)
+      if (timeout.current) {
```

Comment:
is this something achievable using `import { debounce } from 'lib/debounce'`?

Reply:
The `debounce` function allows you to create an unique function to be used in each debounce. The problem with this case is that the function will need the latest version of the `changeFilter` function, which changes each time the path changes, thus making it impossible to have an unique debounce function.